src/components/material/material-buttons/online/MaterialButtonOnlineDigitalArticle.tsx

```diff
@@ -17,6 +20,15 @@ export interface MaterialButtonOnlineDigitalArticleProps {
 const MaterialButtonOnlineDigitalArticle: FC<
   MaterialButtonOnlineDigitalArticleProps
 > = ({ pid, size, dataCy = "material-button-online-digital-article" }) => {
+  const [isResident, setIsResident] = useState<null | boolean>(null);
+  const { isLoading } = useGetPatronInformationByPatronIdV2({
+    enabled: !isAnonymous(),
+    onSuccess: (data) => {
+      if (data?.patron?.resident !== undefined) {
+        setIsResident(data.patron.resident);
+      }
+    }
+  });
```

Comment:
Didn't you talk with Kasper about creating a hook for fetching Patron data if user is logged in? Would make sense to use it here as well.

I think you should also create a function alongside the other user functions called ìsResident()` that looks at the `resident` property of the `Patron` object. We probably need it in other places in the future.